openedx/core/djangoapps/content/search/api.py

```diff
@@ -475,6 +476,31 @@ def index_collection_batch(batch, num_done, library_key) -> int:
                     status_cb(f"Error indexing collection batch {p}: {err}")
             return num_done
 
+        ############## Containers ##############
+        def index_container_batch(batch, num_done, library_key) -> int:
+            docs = []
+            for container in batch:
+                try:
+                    container_metadata = lib_api.ContainerMetadata.from_container(
+                        library_key,
+                        container,
+                    )
+                    doc = searchable_doc_for_container(container_metadata.container_key)
```

Comment:
Maybe we can avoid the `from_container` call here if we add a `get_library_container_usage_key` as we did here:

https://github.com/open-craft/edx-platform/blob/f2946bde3324a99fec8461e636b4ce91ae724423/openedx/core/djangoapps/content_libraries/api/libraries.py#L1712-L1720

I'm not sure we have the `container_type` on the `container` object (I think we should have?).

```suggestion
                    container_key = lib_api.get_library_container_usage_key(
                        library_key,
                        container.key,
                        container.type,  # ????????
                    )
```

Alternatively, we can pass the `container_metadata` along, similar as we have here:

https://github.com/open-craft/edx-platform/blob/8c6cbf02068955c823894c206ba9d2ad9f805630/openedx/core/djangoapps/content/search/documents.py#L491

Reply:
> I'm not sure we have the `container_type` on the container object (I think we should have?).

This is a good question.. @kylecrawshaw @ormsbee should the `Container` model have a `container_type` field (like `Component` has a `component_type`)? This would allow us to create the container locator/usage key from the `Container` object alone.

Reply:
We don't have `container_type` on the container object, and technically it's not necessary because you can check which of the relationships exist - exactly one of `container.unit`, (or in the future) `container.subsection` or `container.section` will be set. If the container is loaded with `select_related` this is pretty efficient and I don't think we need the redundant `container_type` field, but we can always add it on later as an optimization.